### PR TITLE
fixed the built and checked issues

### PR DESCRIPTION
**Pull request by ainilaha**
1. Fixed build issues raised by MsSqlTools outside docker.
2. cleaned up the issues identified by running checks.

### CODE REVIEW

File: DESCRIPTION
Comment (cjendres1) [inline DESCRIPTION:1]: You both should add your info to the Authors field with role = 'aut'

**Review by cjendres1 — CHANGES_REQUESTED** [review]
Made a few minor comments. The one issue that will impact CRAN checks is that you are now calling the DBI library, I believe that DBI needs to be added to the Imports.

File: R/nhanes_tables.R [hunk 1]
```diff
@@ -38,8 +38,8 @@ nhanesTables <- function(data_group, year, nchar=128, details = FALSE, namesonly
     stop("Invalid survey group")
     return(NULL)
   }
-
-  if(!is.na(.collection_date) & !is.na(.container_version)){
+  
+  if(is.numeric(year) & !is.na(.collection_date) & !is.na(.container_version)){
```
[the remark this answers is not in the document]
Comment (ainilaha) [reply R/nhanes_tables.R:43]: I will discuss it with Robert and see whether will include the Pre-pandemic and youth survey in further versions. But, we do not have them in the DB for now.

File: README.md [hunk 2]
```diff
@@ -37,6 +37,8 @@ install_github("cjendres1/nhanes")
 
 **Use nhanesA in Docker**
 
+The Docker container hosts the data, allowing for faster access and manipulation directly from the local Docker environment.
```
[the remark this answers is not in the document]
Comment (ainilaha) [reply README.md]: updated.

File: DESCRIPTION
Comment (ainilaha) [reply DESCRIPTION:1]: Added.

File: R/db_util.R [hunk 3]
```diff
@@ -4,29 +4,20 @@
...
-  removeTaskCallback(which(!after %in% before))
 
-  df <- DBI::dbGetQuery(cn, sql)
-  DBI::dbDisconnect(cn)
-
-  df
+.nhanesQuery = function(sql){
+  if(is.na(.collection_date) | is.na(.container_version)) 
+    stop("can't run .nhanesQuery - no DB detected")
+ return(DBI::dbGetQuery(cn(), sql))
```
[the remark this answers is not in the document]
Comment (ainilaha) [reply R/db_util.R:12]: added into suggests.

**Review by cjendres1 — APPROVED** [review]
This looks good!
We should next discuss routine testing.

**Pull request merged**